client/solve.go

```diff
@@ -90,7 +91,20 @@ func (c *Client) solve(ctx context.Context, def *llb.Definition, runGateway runG
 		return nil, errors.New("invalid with def and cb")
 	}
 
-	syncedDirs, err := prepareSyncedDirs(def, opt.LocalDirs)
+	// merge local mounts and fallback local directories together
+	mounts := make(map[string]fsutil.FS)
+	for k, mount := range opt.LocalMounts {
+		mounts[k] = mount
+	}
+	for k, dir := range opt.LocalDirs {
+		mount, err := fsutil.NewFS(dir)
+		if err != nil {
+			return nil, err
+		}
+		mounts[k] = mount
+	}
```

Comment:
Not blocking on this, but I still think this should be moved to a function and I think the ordering of the merge should be reversed so `LocalMounts` take precedence over `LocalDirs`. If we're going to do the merge, the new option should take precedence over the old option.

```go
func prepareMounts(...) (map[string]fsutil.FS, error) {
    // ...
}
```

Reply:
> I still think this should be moved to a function

I don't particularly mind, updated.

> I think the ordering of the merge should be reversed so LocalMounts take precedence over LocalDirs

I've changed this, so that in case of a key-clash we error out instead of implicitly overriding.

Reply:
I like the error on key-clash better. Seems like a good solution.